src/include/sof/lib/dai-zephyr.h

```diff
@@ -0,0 +1,269 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2016 Intel Corporation. All rights reserved.
+ *
+ * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ *         Keyon Jie <yang.jie@linux.intel.com>
+ */
+
+/**
+  * \file include/sof/lib/dai.h
+  * \brief DAI Drivers definition
+  * \author Liam Girdwood <liam.r.girdwood@linux.intel.com>
+  * \author Keyon Jie <yang.jie@linux.intel.com>
+  */
+
+#ifndef __SOF_LIB_DAI_ZEPHYR_H__
+#define __SOF_LIB_DAI_ZEPHYR_H__
+
+#include <platform/lib/dai.h>
+#include <sof/bit.h>
+#include <sof/list.h>
+#include <sof/lib/io.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/dma.h>
+#include <sof/list.h>
+#include <sof/sof.h>
+#include <sof/spinlock.h>
+#include <sof/trace/trace.h>
+#include <sof/ipc/topology.h>
+#include <sof/audio/pcm_converter.h>
+#include <sof/audio/ipc-config.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <device.h>
+
+/** \brief If the device does not exist it will be created */
+#define DAI_CREAT		BIT(0)
+
+struct dai {
+	uint32_t index;
+	uint32_t type;
+	uint32_t dma_caps;
+	uint32_t dma_dev;
+	const struct device *drv;
+};
+
+struct timestamp_cfg {
+	uint32_t walclk_rate; /* Rate in Hz, e.g. 19200000 */
+	int type; /* SSP, DMIC, HDA, etc. */
+	int direction; /* Playback, capture */
+	int index; /* For SSPx to select correct timestamp register */
+	int dma_id; /* GPDMA id*/
+	int dma_chan_index; /* Used GPDMA channel */
+	int dma_chan_count; /* Channels in single GPDMA */
+};
+
+struct timestamp_data {
+	uint64_t walclk; /* Wall clock */
+	uint64_t sample; /* Sample count */
+	uint32_t walclk_rate; /* Rate in Hz, e.g. 19200000 */
+};
+
+struct timestamp_ops {
+	int (*ts_config)(struct dai *dai, struct timestamp_cfg *cfg);
+	int (*ts_start)(struct dai *dai, struct timestamp_cfg *cfg);
+	int (*ts_stop)(struct dai *dai, struct timestamp_cfg *cfg);
+	int (*ts_get)(struct dai *dai, struct timestamp_cfg *cfg,
+		      struct timestamp_data *tsd);
+};
+
+/**
+ * \brief DAI group information
+ */
+struct dai_group {
+	/**
+	 * Group ID
+	 */
+	uint32_t group_id;
+
+	/**
+	 * Number of DAIs in this group
+	 */
+	uint32_t num_dais;
+
+	/**
+	 * Number of DAIs to receive a trigger before processing begins
+	 */
+	uint32_t trigger_counter;
+
+	/**
+	 * Trigger command to propagate
+	 */
+	int trigger_cmd;
+
+	/**
+	 * Last trigger error
+	 */
+	int trigger_ret;
+
+	/**
+	 * Group list
+	 */
+	struct list_item list;
+};
+
+/**
+ * \brief DAI runtime data
+ */
+struct dai_data {
+	/* local DMA config */
+	struct dma_chan_data *chan;
+	uint32_t stream_id;
+	struct dma_sg_config config;
```

Comment:
this structure needs aligned to Zephyr too, i.e. dma_sg_config is not needed, likewise not sure if dma_chan+data is needed.